File: dotnet/src/xjustiz.core-dotnet/Models/Entities/Core/XjustizFachspezifischeDatenCore.cs
namespace xjustiz.core_dotnet.Models.Entities.Core;

using System.Xml.Serialization;
using xjustiz.core_dotnet.Models.Helpers;
using xjustiz.core_dotnet.Util.Versioning;

/// <summary>
/// Fachspezifische Daten eines Dokuments.<br/>
/// <u><b>Subject-specific data:</b></u> Specialized data of a document.
/// </summary>
[XJustizCoreAvailability(XJustizCoreVersion.V0_2_0)]
public class XjustizFachspezifischeDatenCore : XjustizFachspezifischeDaten
{
    /// <summary>
    /// Repräsentiert eine Datei.<br/>
    /// <u><b>File:</b></u> Represents a file.
    /// </summary>
    [XmlElement("datei", Namespace = SchemeConstants.XJustizCore_Tns)]
    [XJustizCoreAvailability(XJustizCoreVersion.V0_2_0)]
    public new DateiCore Datei { get; set; } = new();
}

File: sdk/java/src/main/java/de/xjustiz/core/models/XjustizFachspezifischeDatenCore.java
package de.xjustiz.core.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.xml.bind.annotation.XmlAccessType;
import jakarta.xml.bind.annotation.XmlAccessorType;
import jakarta.xml.bind.annotation.XmlType;

/**
 * Fachspezifische Daten eines Dokuments.
 * <p>
 * <u><b>Subject-specific data:</b></u> Specialized data of a document.
 */
@XmlType
@XmlAccessorType(XmlAccessType.FIELD)
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class XjustizFachspezifischeDatenCore extends XjustizFachspezifischeDaten {

    /**
     * Default constructor.
     */
    public XjustizFachspezifischeDatenCore() {}

}